calyx/src/passes/group_to_invoke.rs

```diff
@@ -31,100 +34,190 @@ impl Named for GroupToInvoke {
     }
 }
 
+// Returns true if port's parent is cell. We can do this by checking if they
+// have the same name.
+fn cell_is_parent(port: &ir::Port, cell: &ir::RRC<ir::Cell>) -> bool {
+    port.get_parent_name() == cell.borrow().name()
```

Comment:
This should use [`Rc::ptr_eq`](https://doc.rust-lang.org/stable/std/rc/struct.Rc.html#method.ptr_eq) to check that the parent pointer of the port is the same as the provided cell. You can match on the `.parent` field and see if the `uprade()` call to the parent pointer generates the same pointer